plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/refactoring/preview/PreviewPresenter.java

```diff
@@ -108,94 +83,218 @@ public void setTitle(String title) {
   /** {@inheritDoc} */
   @Override
   public void onCancelButtonClicked() {
-    EditorPartPresenter activeEditor = editorAgent.getActiveEditor();
-    if (activeEditor instanceof TextEditor) {
-      ((TextEditor) activeEditor).setFocus();
-    }
+    view.close();
   }
 
   /** {@inheritDoc} */
   @Override
   public void onAcceptButtonClicked() {
-    clientServerEventService
-        .sendFileTrackingSuspendEvent()
-        .then(
-            success -> {
-              applyRefactoring();
-            });
-  }
+    updateFinalEdits();
+
+    WorkspaceEdit edit = new WorkspaceEdit();
+    edit.setChanges(workspaceEdit.getChanges());
+    edit.setResourceChanges(new LinkedList<>());
+    for (ResourceChange resourceChange : workspaceEdit.getResourceChanges()) {
+      edit.getResourceChanges().add(Either.forLeft(resourceChange));
+    }
 
-  private void applyRefactoring() {
-    refactoringService
-        .applyRefactoring(session)
-        .then(
-            refactoringResult -> {
-              List<ChangeInfo> changes = refactoringResult.getChanges();
-              if (refactoringResult.getSeverity() == OK) {
-                view.close();
-                refactoringUpdater
-                    .updateAfterRefactoring(changes)
-                    .then(
-                        refactoringUpdater
-                            .handleMovingFiles(changes)
-                            .then(clientServerEventService.sendFileTrackingResumeEvent()));
-              } else {
-                view.showErrorMessage(refactoringResult);
-                refactoringUpdater
-                    .handleMovingFiles(changes)
-                    .then(clientServerEventService.sendFileTrackingResumeEvent());
-              }
-            });
+    applyWorkspaceEditAction.applyWorkspaceEdit(edit);
+    view.close();
+    refactoringActionDelegate.closeWizard();
```

Comment:
It invokes `onCancelButtonClicked();`
I think `view.close()` is enough.

Reply:
`view.close()` closes just Preview window, but also we need to close the other one (Rename or Move) window which was opened previously